lib/Sema/CSApply.cpp

```diff
@@ -4756,6 +4763,14 @@ Expr *ExprRewriter::coerceTupleToTuple(Expr *expr,
     convertedElts.emplace_back(toEltType, toLabel, ParameterTypeFlags());
   }
 
+  // Shuffling tuple elements is an anti-pattern worthy of a diagnostic.  We
+  // will form the shuffle for now, but a future compiler should decline to
+  // do so and begin the process of removing them altogether.
+  if (anythingShuffled) {
+    cs.getASTContext().Diags.diagnose(
```

Comment:
Nit: The `ASTContext` is already accessible via `ctx` at the top of this method.